### PR TITLE
LPS-143416 FDS browser console errors

### PR DESCRIPTION
**Pull request by thektan**
https://issues.liferay.com/browse/LPS-143416

Most errors were fixed by https://issues.liferay.com/browse/LPS-148447, but in the Blueprints admin page, the "copy" dropdown action uses a `post` request method so I'm updating the prop types to allow other possible methods.

I also noticed the `target` prop should allow more types of modals since there is code supporting more `modal-*` targets https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js#L62-L75

### CODE REVIEW

**Comment by thektan** [comment]
ci:test:sf

**Comment by thektan** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
ci:test:relevant

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js [hunk 1]
```diff
@@ -430,22 +430,26 @@ ActionsDropdownRenderer.propTypes = {
...
 				'async',
 				'headless',
 				'inlineEdit',
+				'link',
+				'modal',
+				'modal-full-screen',
+				'modal-lg',
+				'modal-permissions',
+				'modal-sm',
+				'sidePanel',
```
Comment (diegonvs) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:452]: Hey @markocikos, why we are typing with oneOf this string? I think PropTypes.string is enough

Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:452]: > why we are typing with oneOf this string? I think PropTypes.string is enough

I strongly dislike that this `target` prop even exists. It's basically a code constant that you pass in and it decides which one of internal implementation is applied. This is bad design for a general purpose component. For one thing, API should be descriptive, for example `inlineEdit` boolean. More importantly, specific implementations should be left to specific usages. We should analyze how these are used. Those who are rarely used, we should move out of the component and into usages, set through `propsTransformer`. The same for those that implement product-specific logic, like `modal-permissions`. Modal size is something that should be defined in the specific usage.

A list of these codes can be helpful to us for this migration, that's the only reason I would keep it. I think that changing this to `PropTypes.string` would also be fine, I don't think it'll be a big problem for us to to find our way through code.

**Review by markocikos — APPROVED** [review]
LGTM!

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**